Source/DiabloUI/ui_item.h

```diff
@@ -42,36 +43,60 @@ enum UiFlags : uint16_t {
 	// clang-format on
 };
 
+inline UiFlags operator|(UiFlags lhs, UiFlags rhs)

```

Comment:
@AJenbo had a [comment](https://github.com/diasurgical/devilutionX/pull/1869#discussion_r626852934) for this kind of bitflag operators.
Perhaps this refactoring PR is a good point to address the comment.

Reply:
Ooh, thanks for pointing that out. (I think I actually read that comment trail and then remembered nothing from it when I was looking at this 😓 )

I was looking at C++20 constraints, which would provide for something like the following (if we could even target that version of the language):
```cpp
template<typename T> requires std::is_enum_v<T>
inline T operator|(T lhs, T rhs)
{
	using V = std::underlying_type_t<T>;
	return static_cast<T>(static_cast<V>(lhs) | static_cast<V>(rhs));
}
```
Even if we could use C++20, C++ enums don't have any distinction between flags types and non-flags types like you get in other languages so that would allow for some nonsensical operations like `Direction::DIR_S | Direction::DIR_N`.

The grisumbras/enum-flags library you found in that comment looks ideal, that wrapper class provides actual friend functions and a way to mark certain enums as being used for flags.